### DISCUSSION TITLE
Admin and user dashboard automatically generated

### DISCUSSION THREAD

**Discussion by matijaSos** [wasp-lang/wasp#1386] [Archived]
This could be a really cool feature IMO. Django is famous for its admin dashboard and we could do the same, and make it regenerate with every change. The same goes for user dashboard - we could show number of users, even some of their activity etc.

**Comment by Zeko369** [comment]
Automatically generated sounds a lot like Prisma studio, so for clarity I'd rename this to "configurable built in admin dashboard" or something along those lines, since you almost never want the "autogenerated" dashboard, you always want to remove/add columns, show or not show relations...

(by autogenerated it sounds like `dashboard: true`, what I'd want is a config per entity with ability to use custom functions for rows / forms / stuff like that, so essentially django-admin or administrate gem)

https://docs.djangoproject.com/en/4.2/ref/contrib/admin
https://github.com/thoughtbot/administrate
https://marmelab.com/react-admin

While all these solutions support super simple admin creation, they all require you to define what you want, and that's what I think we should have. Since for just the regular auto generated DB inferred dashboard admins can just run Prisma studio on fly behind a http-auth proxy and that's it 😅

**Reply by Zeko369** [reply]
Also: since wasp is a DSL the handover of the base vs custom pages/functions can be done a lot cleaner than with django-admin (where you define a class with just a `pass` in it and then register it) + LSP can be used to autogenerate the autocomplete for entities / columns / bla

**Comment by Umbrien** [comment]
Wow, it's incredible how I did not think of this before!
It will be a very cool feature. Many sites need it and as of now we have to implement it by ourselves in every React project which is a pain to do, we want to focus on our business logic and user's application, not on admin. Prisma Studio is a minimal admin dashboard for now 😄

I've used Django and its admin dashboard is really good, it even was a reason to choose Django over other technologies for some projects.
But please make it easily extendible as I heard from my friend he had to dig out into Django core sources to implement some highly customized components in admin.

Not sure about a separate user dashboard, anyways it will only be accessible to admins. Yeah, it serves a different purpose, view statistics instead of managing DB data.  Maybe then separate by roles even more, give the option to have a role that can view the user dashboard but can't access the rest of the admin dashboard?
Also, I think "User" dashboard is not the best naming as it is not made for users but for someone between user and admin

**Comment by Martinsos** [comment]
So just to recap -> it would be super cool to have an admin dashboard that Wasp easily generates for you. Not just for users necessarily, but for all the entities. But, the real value comes from being able to customize it in Wasp, so that it can be a custom dashboard, adapted to this specific app (you could pick views you want to see, columns, rows, some kind of analytics, ... . Because otherwise Prisma studio is enough, it already gives you basic data CRUD. So kind of like adjustable Prisma studio.

**Comment by Martinsos** [comment]
Created an issue for this one: https://github.com/wasp-lang/wasp/issues/1420 .
